src/EditorFeatures/VisualBasic/Formatting/Indentation/SpecialFormattingOperation.vb

```diff
@@ -106,6 +106,62 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.Formatting.Indentation
                     baseToken, startToken, endToken, TextSpan.FromBounds(baseToken.Span.End, closeBrace.Span.End), indentationDelta, IndentBlockOption.RelativePosition))
         End Sub
 
+        Private Sub AddArgumentListIndentBlockOperations(operations As List(Of IndentBlockOperation), node As SyntaxNode)
+            Dim argumentList = TryCast(node, ArgumentListSyntax)
+            If argumentList Is Nothing OrElse
+               argumentList.OpenParenToken.IsMissing OrElse
+               argumentList.Arguments.Count = 0 OrElse
+               argumentList.Arguments.All(Function(a) a.IsMissing) Then
+                Return
+            End If
+
+            Dim openBrace = argumentList.GetFirstToken(includeZeroWidth:=True)
+            Dim closeBrace = argumentList.GetLastToken(includeZeroWidth:=True)
+
+            Dim sourceText = node.SyntaxTree.GetText()
+
+            ' First, create a list of all arguments that start lines.
+            Dim arguments = New List(Of ArgumentSyntax)
+
+            Dim previousLineNumber = -1
+            For Each argument In argumentList.Arguments
+                Dim lineNumber = sourceText.Lines.GetLineFromPosition(argument.SpanStart).LineNumber
+                If lineNumber > previousLineNumber Then
+                    arguments.Add(argument)
```

Comment:
❓ What happens if the argument is split across lines?

```vb
Method(a +
  b, c +
  d) ' <-- What happens if you add , and hit enter here?
```

Reply:
I am fine to be better on smart indent on expression. but just as FYI.

both C# and VB never tries to be too smart on multi line expressions. every one has different liking on those on every different cases even for same construct.

we do have some special cases like argument or parameter lists, but even for those, we did very simple thing.

most of time, if expression is more than 2 lines a part, we just give up and act like simple block indent (either align with previous line or 4 space added  after previous line)

Reply:
one thing I hope we don't do is, we forcing people to follow our liking and make them to depend options for it.

Reply:
Thanks! Good question. I'll add a test or two to cover this scenario.

Reply:
@heejaechang: We aren't forcing people to follow our liking with this change and no options would ever be required. Instead, we're treating the special case of argument lists (which is different between C# and VB) to be a bit more natural.